### DISCUSSION TITLE
Why is X5000 and X6000 combined?

### DISCUSSION THREAD

**Discussion by ovoME** [ChefKissInc/NootedRed#264] [Q&A] [answered]
(I must be a rookie of hackintosh)
I found that the kext uses x5000 and x6000 drivers in macOS
but for raven based cpus, the apu inside uses gcn5, the same architecture as vega56/64, and they uses AMD10000Controller.kext in macOS
So why NRed  utilize x5/6000 driver instead of utilizing gcn5 driver?
Hope to get the answer! Thank you very much!

**Comment by VisualEhrmanntraut** [comment, marked as answer]
X5000 is GCN 5. I combined X6000 purely for the display because the Vega dGPUs use a different display architecture, DCE (Display Controller Engine), while the Vega iGPUs use DCN (Display Core Next). I plan to remove the dependency on AMDRadeonX6000 with reverse engineered code, but AMDRadeonX6000Framebuffer will remain. In older versions the VCN engine was also snatched but it didn't work properly or at all so I removed it.
